fix: RHTAP-3416 'template' command uses the wrong namespace

Roming22 opened the pull request:
- The chart release namespace is retrieved from the config.yaml.
- Helm connects to the server to provide an accurate rendering of the templates.

rh-pre-commit.version: 2.3.2
rh-pre-commit.check-secrets: ENABLED

otaviof commented:
/retest

otaviof commented on pkg/deployer/helm.go:
```diff
@@ -68,7 +68,7 @@ func (h *Helm) helmInstall(vals chartutil.Values) (*release.Release, error) {
 	c.ClientOnly = h.flags.DryRun
 	if h.flags.DryRun {
-		c.DryRunOption = "client"
+		c.DryRunOption = "server"
```
Why are we changing this from `client` to `server`?

Roming22 replied:
Because a 'client' rendering is mostly useless for troubleshooting. The output will not be representative of what will be deployed on the cluster.

Switching to 'server' allows Helm to contact the server and produce the manifests as they will be applied at deployment time.

otaviof commented on installer/charts/rhtap-dh/values.yaml:
```diff
@@ -1,4 +1,7 @@
 ---
+debug:
+  ci: false
+
```
Is the `rhtap-dh` Chart using `.debug.ci` in the templates? If I remember correctly, the other charts don't have this attribute defined.

Roming22 replied:
That's the only template that uses that flag.
I've added it because it allows to call `helm template` on the chart without any error.

otaviof commented on pkg/config/config.go:
```diff
@@ -55,6 +55,17 @@ func (c *Config) GetBaseDir() string {
 }
 
+// GetChart returns a dependency chart configuration.
+func (c *Config) GetChart(logger *slog.Logger, chart string) (*Dependency, error) {
```
Given this method returns a `&Dependency{}`, let's name it `GetDependency` instead?

Roming22 replied:
With your latest changes, I think I can rollback that change.

Roming22 replied:
It is in fact still needed so I've renamed it.

otaviof reviewed (changes requested):
Let me understand the reasons behind the dry-run changes, please.

xinredhat commented:
/lgtm

Roming22 committed:
fix: RHTAP-3416 'template' command uses the wrong namespace

- The chart release namespace is retrieved from the config.yaml.
- Helm connects to the server to provide an accurate rendering of the
  templates.

rh-pre-commit.version: 2.3.2
rh-pre-commit.check-secrets: ENABLED

Roming22 committed:
fix: support absolute paths for --config

rh-pre-commit.version: 2.3.2
rh-pre-commit.check-secrets: ENABLED

Roming22 commented:
@otaviof I've fixed your comment, and answered your questions. I've also added an extra commit to fix an issue created by a recent commit.

PTAL.

xinredhat commented:
/retest

Roming22 commented:
/retest

otaviof commented on pkg/deployer/helm.go:
```diff
@@ -68,7 +68,7 @@ func (h *Helm) helmInstall(vals chartutil.Values) (*release.Release, error) {
 	c.ClientOnly = h.flags.DryRun
 	if h.flags.DryRun {
-		c.DryRunOption = "client"
+		c.DryRunOption = "server"
```
By using `client` means Helm dry-run mode validates configurations locally, while `server` dry-run mode sends the request to the API server for full validation without persisting changes. We can improve this later, let's highlight that 👍

otaviof reviewed (approved):
/lgtm

With a single remark about the dry-run mode 😊
